source/tests/script.test.js

```diff
@@ -1,32 +1,58 @@
+import '../src/scripts/script';
+import '../src/components/TaskItem';
+
+import { addTemplates, dispatchDOMLoadedEvent } from './utils';
+import { TASK_ITEM_TEMPLATE } from './Constants';
+
 window.HTMLMediaElement.prototype.play = () => { /* do nothing */ };
-beforeEach(() => {
-    require('../src/scripts/script');
-    require('../src/components/TaskItem');
-    localStorage.setItem('tasks', '[]');
-    localStorage.setItem('id', '0');
-    localStorage.setItem('theme', 'light');
-    localStorage.setItem('volume', 50);
-    localStorage.setItem('state', 'default');
-    document.body.innerHTML = `
+
+let pageTemplate;
+
+beforeAll(async () => {
+    const templates = await addTemplates([
+        TASK_ITEM_TEMPLATE,
+    ], __dirname);
+
+    // Page template to be used for every test
+    // Initialized here to prevent multiple string building
+    pageTemplate = `
+        ${templates}
         <ul id="task-list-elements">
-        </ul>   
+        </ul>
         <div id='focus-task'>
             <h2 id='select-focus'></h2>
-        </div>   
+        </div>
     `;
 });
 
+beforeEach(() => {
+    document.body.innerHTML = pageTemplate;
+});
+
 afterEach(() => {
     localStorage.clear();
 });
 
+/**
+ * !! Due to how script.js is written, this is necessary to do before running
+ * tests that manipulate the localStorage. This is because the tests relied on
+ * beforeEach initializing localStorage manually and used the event listener
+ * to populate the document with the tasks.
+ *
+ * We should aim to no longer need this function and to unlink the localStorage
+ * manipulation with the DOMContentLoaded event listener in script.js
+ */
```

Comment:
Do we aim for initializing the localStorage once and run all the tests or something else?

Reply:
script.js currently waits for the DOMContentLoaded event and if individual value from the initializeLocalStorage function is undefined, it will reset all of the values. This was mitigated by beforeEach initializing the localStorage and then passing the DOMContentLoaded event. This PR was already quite hefty so I didn't fix it here since I wanted a single purpose PR. 

This can be fixed many ways:
1. As you said, we can make it initialize localStorage once and then run all the tests, but by moving localStorage initialization out of the DOMContentLoaded event listener. Localstorage is immediately available as the page begins loading meaning there is no need to use the DOMContentLoaded on default initializers.
2. Change the default initializer in script.js to a one-by-one instead of an all-for-nothing. This has the added benefit of any developers adding in new settings and preventing complete reset of their settings. This isn't that much of an issue for the course since we're likely the last people to work on this project, but showing that we have the future development mentality doesn't hurt.